packages/wrangler/src/api/pages/create-worker-bundle-contents.ts

```diff
@@ -0,0 +1,64 @@
+import { readFileSync } from "node:fs";
+import path from "node:path";
+import { Response } from "undici";
+import { createWorkerUploadForm } from "../../create-worker-upload-form";
+import type { BundleResult } from "../../bundle";
+import type { CfWorkerInit } from "../../worker";
+import type { Blob } from "node:buffer";
+import type { FormData } from "undici";
+
+/**
+ * Takes a Worker bundle - `BundleResult` - and generates the contents
+ * of the upload worker bundle file
+ */
+export async function createUploadWorkerBundleContents(
+	workerBundle: BundleResult
+): Promise<Blob> {
+	const workerBundleFormData = createWorkerBundleFormData(
+		workerBundle as BundleResult
+	);
+	return await new Response(workerBundleFormData).blob();
+}
+
+/**
+ * Creates a `FormData` upload from a `BundleResult`
+ */
+function createWorkerBundleFormData(workerBundle: BundleResult): FormData {
+	const mainModule = {
+		name: path.basename(workerBundle.resolvedEntryPointPath),
+		content: readFileSync(workerBundle.resolvedEntryPointPath, {
+			encoding: "utf-8",
+		}),
+		type: workerBundle.bundleType || "esm",
+	};
+
+	const worker: CfWorkerInit = {
+		name: mainModule.name,
+		main: mainModule,
+		modules: workerBundle.modules,
+		bindings: {
+			vars: undefined,
+			kv_namespaces: undefined,
+			wasm_modules: undefined,
+			text_blobs: undefined,
+			data_blobs: undefined,
+			durable_objects: undefined,
+			queues: undefined,
+			r2_buckets: undefined,
+			d1_databases: undefined,
+			services: undefined,
+			analytics_engine_datasets: undefined,
+			dispatch_namespaces: undefined,
+			logfwdr: undefined,
+			unsafe: undefined,
+		},
+		migrations: undefined,
+		compatibility_date: undefined,
+		compatibility_flags: undefined,
+		usage_model: undefined,
+		keepVars: undefined,
+		logpush: undefined,
+	};
+
+	return createWorkerUploadForm(worker);
```

Comment:
@GregBrimble regarding us stripping away those `bindings` from `metadata`... It seems that the `WorkerMetadata` interface is very specific in requiring those bindings to be present 👇 

```
export interface WorkerMetadata {
	main_module?: string;
	body_part?: string;
	compatibility_date?: string;
	compatibility_flags?: string[];
	usage_model?: "bundled" | "unbound";
	migrations?: CfDurableObjectMigrations;
	capnp_schema?: string;
	bindings: WorkerMetadataBinding[];
	keep_bindings?: WorkerMetadataBinding["type"][];
	logpush?: boolean;
}
```
I can still try to strip them from our `metadata` before we send them further off into the world ;). Unless we think we might very possible need them as well at some point?

Reply:
I think it would be nice if we could omit the empty `"bindings":[]`, but totally not critical and we can do it in cleanup if we still care enough.

Reply:
nah, let's do this now. You know how cleanup works 😋

Reply:
✅